ubcpi/ubcpi.py

```diff
@@ -0,0 +1,468 @@
+                    'display_name': self.display_name,
+                    'correct_answer': self.correct_answer,
+                    'correct_rationale': self.correct_rationale,
+                    'rationale_size': self.rationale_size,
+                    'question_text': self.question_text,
+                    'options': self.options,
+                    'algo': self.algo,
+                    'algos': {'simple': 'System will select one of each option to present to the students.',
+                              'random': 'Completely random selection from the response pool.'},
+                    'image_position_locations': {
+                    	'above': 'Appears above',
+                    	'below': 'Appears below'
+                    },
+                    'seeds': self.seeds,
+        })
+
+        return frag
+
+    @XBlock.json_handler
+    def studio_submit(self, data, suffix=''):
+        self.display_name = data['display_name']
+        self.question_text = data['question_text']
+        self.rationale_size = data['rationale_size']
+        self.options = data['options']
+        self.correct_answer = data['correct_answer']
+        self.correct_rationale = data['correct_rationale']
+        self.algo = data['algo']
+        self.seeds = data['seeds']
+
+        return {'success': 'true'}
+
+    @staticmethod
+    def resource_string(path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    @XBlock.handler
+    def get_asset(self, request, suffix=''):
+        filename = request.params.get('f')
+        return Response(self.resource_string('static/js/partials/' + filename), content_type='text/html')
+
+    @classmethod
+    def get_base_url_path_for_course_assets(cls, course_key):
+        """
+        Slightly modified version of StaticContent.get_base_url_path_for_course_assets.
+        Code is copied as we don't want to introduce the dependency of edx-platform
+        :param course_key: CourseKey
+        :return: course asset base URL string
+        """
+        if course_key is None:
+            return None
+
+        # assert isinstance(course_key, CourseKey)
+        placeholder_id = uuid.uuid4().hex
+        # create a dummy asset location with a fake but unique name. strip off the name, and return it
+        url_path = cls.serialize_asset_key_with_slash(
+            course_key.make_asset_key('asset', placeholder_id).for_branch(None)
+        )
+        return url_path.replace(placeholder_id, '')
+
+    @staticmethod
+    def serialize_asset_key_with_slash(asset_key):
+        """
+        Legacy code expects the serialized asset key to start w/ a slash; so, do that in one place
+        :param asset_key:
+        """
+        url = unicode(asset_key)
+        if not url.startswith('/'):
+            url = '/' + url  # TODO - re-address this once LMS-11198 is tackled.
```

Comment:
This is actually from edx platform: https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/contentstore/content.py#L168. I couldn't find a documented way to generate the asset URL in xblock and didn't get any answer from IRC. So I dogged the edx-platform code and found those are used.